ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_8/ActiveContractsServiceIT.scala

```diff
@@ -743,14 +743,18 @@ class ActiveContractsServiceIT extends LedgerTestSuite {
     enabled = _.acsActiveAtOffsetFeature,
     disabledReason = "Requires ACS with active_at_offset",
     partyAllocation = allocate(SingleParty),
+    runConcurrently = false,
   )(implicit ec => { case Participants(Participant(ledger, party)) =>
     val transactionFilter = Some(TransactionFilter(filtersByParty = Map(party.unwrap -> Filters())))
     for {
+      ledgerEndOffset <- ledger.currentEnd()
+      // We append "11" to obtain a new offset that is guaranteed to be larger that the current ledger end
+      offsetBeyondLedgerEnd = ledgerEndOffset.getAbsolute + "11"
```

Comment:
I see that we actually already have a `ParticipantTestContext.offsetBeyondLedgerEnd()` method. It prepends `FFFF` instead of appending, but it should be fine as well. Can we just use this one?

Reply:
Interesting with `offsetBeyondLedgerEnd`'s original impl I'm getting
```
- [ActiveContractsServiceIT:ActiveAtOffsetAfterLedgerEnd] Fail when active_at_offset is after the ledger end offset ... Assertion failed
  Actual error id (NON_HEXADECIMAL_OFFSET) does not match expected error id (OFFSET_AFTER_LEDGER_END}. Actual error: io.grpc.StatusRuntimeException: INVALID_ARGUMENT: NON_HEXADECIMAL_OFFSET(8,0): Offset in active_at_offset not specified in hexadecimal: FFFF000000000000223d: Reason: cannot parse HexString FFFF000000000000223d
```
After I changed it to prepend lowercase `ffff` it passes

Reply:
Interesting. This method is actually used in other tests as well that test `LedgerApiErrors.RequestValidation.OffsetAfterLedgerEnd` (e.g. see `CSCAfterEnd`). It means the request validation wrt offset is not consistent across these endpoints?

Reply:
FYI @mziolekda probably this is something to have on the tech debt log

Reply:
I did one more experiment.
I created an offset which is a valid ledger string but not a hex string and is before the ledger end when you run `CommandSubmissionCompletionIT:CSCAfterEnd` individually:
```
LedgerOffset(LedgerOffset.Value.Absolute("000000000000220:"))
```

This resulted in client receiving internal error and this was in the participant logs:
```
java.lang.IllegalArgumentException: cannot parse HexString 000000000000220:
	at com.daml.platform.ApiOffset$.tryFromString(ApiOffset.scala:19)
	at com.daml.platform.index.IndexServiceImpl.$anonfun$convertOffset$1(IndexServiceImpl.scala:423)
	at com.daml.platform.index.IndexServiceImpl.getCompletions(IndexServiceImpl.scala:203)
	at com.daml.platform.apiserver.TimedIndexService.$anonfun$getCompletions$1(TimedIndexService.scala:84)
	at com.daml.metrics.Timed$.source(Timed.scala:59)
	at com.daml.platform.apiserver.TimedIndexService.getCompletions(TimedIndexService.scala:84)
	at com.daml.platform.apiserver.services.ApiCommandCompletionService.$anonfun$completionStreamSource$4(ApiCommandCompletionService.scala:69)
	at com.daml.logging.LoggingContext$.withEnrichedLoggingContext(LoggingContext.scala:60)
	at com.daml.platform.apiserver.services.ApiCommandCompletionService.$anonfun$completionStreamSource$3(ApiCommandCompletionService.scala:62)
	at scala.util.Either.fold(Either.scala:189)
	at com.daml.platform.apiserver.services.ApiCommandCompletionService.$anonfun$completionStreamSource$1(ApiCommandCompletionService.scala:58)
	at akka.stream.impl.fusing.Map$$anon$1.onPush(Ops.scala:52)
	at akka.stream.impl.fusing.GraphInterpreter.processPush(GraphInterpreter.scala:542)
	at akka.stream.impl.fusing.GraphInterpreter.processEvent(GraphInterpreter.scala:528)
	at akka.stream.impl.fusing.GraphInterpreter.execute(GraphInterpreter.scala:390)
	at akka.stream.impl.fusing.GraphInterpreterShell.runBatch(ActorGraphInterpreter.scala:650)
	at akka.stream.impl.fusing.ActorGraphInterpreter$SimpleBoundaryEvent.execute(ActorGraphInterpreter.scala:61)
	at akka.stream.impl.fusing.ActorGraphInterpreter$SimpleBoundaryEvent.execute$(ActorGraphInterpreter.scala:57)
	at akka.stream.impl.fusing.ActorGraphInterpreter$RequestMore.execute(ActorGraphInterpreter.scala:289)
	at akka.stream.impl.fusing.GraphInterpreterShell.processEvent(ActorGraphInterpreter.scala:625)
	at akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$processEvent(ActorGraphInterpreter.scala:800)
	at akka.stream.impl.fusing.ActorGraphInterpreter$$anonfun$receive$1.applyOrElse(ActorGraphInterpreter.scala:818)
	at akka.actor.Actor.aroundReceive(Actor.scala:537)
	at akka.actor.Actor.aroundReceive$(Actor.scala:535)
	at akka.stream.impl.fusing.ActorGraphInterpreter.aroundReceive(ActorGraphInterpreter.scala:716)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:580)
	at akka.actor.ActorCell.invoke(ActorCell.scala:548)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:270)
	at akka.dispatch.Mailbox.run(Mailbox.scala:231)
	at akka.dispatch.Mailbox.exec(Mailbox.scala:243)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183
```

Reply:
luckily the offsets are opaque to the clients and they never use offsets that haven been dispensed by the LAPI to begin with. We have a self made problem in the LAPI test tool, that isn't easy to fix given that the old ledger api test tools must work against cranked-up ledger validations. https://digitalasset.atlassian.net/browse/DPP-1387